autoload/go/job.vim

```diff
@@ -91,14 +91,22 @@ function! go#job#Options(args)
   " do nothing in state.complete by default.
   function state.complete(job, exit_status, data)
     if has_key(self, 'custom_complete')
-      let l:winid = win_getid(winnr())
       " Always set the active window to the window that was active when the job
       " was started. Among other things, this makes sure that the correct
       " window's location list will be populated when the list type is
       " 'location' and the user has moved windows since starting the job.
-      call win_gotoid(self.winid)
+      " Make sure to jump only if 'for' was not set explicitly to '_', which
+      " is an indicator that the process will show a list
+      if has_key(self, 'for') && self.for != '_'
```

Comment:
You should be able to use `get()` here, which works like in Python:

    if get(self, 'for', '') != ''

It will return the third parameter (`''`) if the `for` key doesn't exist.

More readable, IMHO.

Same below.

Reply:
Good call. However I've change it to use `_`, because otherwise this wouldn't work. We shouldn't call the win commands in case it's also set to `_` which is the case when you don't want to use `for`